src/modules/storage/LedgerStorage.ts

```diff
@@ -671,6 +729,135 @@ export class LedgerStorage extends Storages {
         });
     }
 
+    /**
+     * Puts proposal result
+     * @param block: The instance of the `Block`
+     */
+    public putProposalResult(block: Block): Promise<void> {
+        function get_proposal_result(
+            storage: LedgerStorage,
+            proposal_id: string,
+            block_validators: IProposalValidator[],
+            voting_end_height: number
+        ): Promise<Boolean> {
+            return new Promise<Boolean>(async (resolve, reject) => {
+                try {
+                    let approved: number = 0;
+                    let opposed: number = 0;
+                    let abstained: number = 0;
+                    let rejected: number = 0;
+                    let quorum: number = 0;
+                    let Qn: number = 0;
+                    let satisfyQuorum: boolean = false;
+                    let F;
+                    let result: boolean = false;
+                    let voting_result: boolean = false;
+                    let votes = await storage.getProposalVotes(proposal_id);
+                    let totoalValidators = await storage.getValidatorsAPI(null, null);
```

Comment:
Height parameters are required.
The height is the height of the block containing proposal data transaction.
Voters are determined by that height.